csharp/ql/lib/semmle/code/csharp/Preprocessor.qll

```diff
@@ -208,21 +208,40 @@ class HiddenLineDirective extends LineDirective {
   override string getAPrimaryQlClass() { result = "HiddenLineDirective" }
 }
 
+abstract private class NumericOrSpanLineDirective extends LineDirective {
+  /** Gets the referenced file of this directive. */
+  File getReferencedFile() { directive_line_file(this, result) }
+}
+
 /**
- * A numeric `#line` directive, such as `#line 200 file`
+ * A numeric `#line` directive, such as `#line 200 file`.
  */
-class NumericLineDirective extends LineDirective {
+class NumericLineDirective extends NumericOrSpanLineDirective {
   NumericLineDirective() { directive_lines(this, 2) }
 
   /** Gets the line number of this directive. */
   int getLine() { directive_line_value(this, result) }
 
-  /** Gets the referenced file of this directive. */
-  File getReferencedFile() { directive_line_file(this, result) }
-
   override string getAPrimaryQlClass() { result = "NumericLineDirective" }
 }
 
+/**
+ * A line span `#line` directive, such as `#line (1, 1) - (3, 10) 5 file`.
+ */
+class SpanLineDirective extends NumericOrSpanLineDirective {
+  SpanLineDirective() { directive_lines(this, 3) }
+
+  /** Gets the offset of this directive. */
+  int getOffset() { directive_line_offset(this, result) }
+
+  /** Gets the span of this directive. */
+  predicate span(int startLine, int startColumn, int endLine, int endColumn) {
+    directive_line_span(this, startLine, startColumn, endLine, endColumn)
+  }
```

Comment:
The comment doesn't match the predicate. `/** Holds if the specified start and end positions match this SpanLineDirective. */`

Reply:
Ah yes - you are right! I will fix that!